### PR TITLE
Multiple app support

### LINKED ISSUE #3072: Support multiple Realm Apps

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

Add support for multiple Realm Apps.

This closes #3072 

## ☑️ ToDos

* [x] 📝 Changelog entry
~* [ ] 📝 `Compatibility` label is updated or copied from previous entry~
* [x] 🚦 Tests
~* [ ] 📝 Public documentation PR created or is not necessary~
~* [ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
~* [ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: src/js_realm.hpp [hunk 1]
```diff
@@ -831,12 +834,16 @@ void RealmClass<T>::get_is_closed(ContextType ctx, ObjectType object, ReturnValu
...
+    if (config.sync_config) {
+        auto user = config.sync_config->user;
+        if (user) {
+            if (std::shared_ptr<SyncSession>session = user->sync_manager()->get_existing_active_session(config.path)) {
+                return return_value.set(create_object<T, SessionClass<T>>(ctx, new WeakSession(session)));
+            }
+        }
     }
-
+    return_value.set(Value::from_null(ctx));
```
Comment (blagoev) [inline src/js_realm.hpp:846]: could use return_value.set_null();

File: src/node/node_class.hpp [hunk 2]
```diff
@@ -328,7 +328,6 @@ WrappedObject<ClassType>::WrappedObject(const Napi::CallbackInfo& info) : Napi::
 	catch (const node::Exception& e) {
 		Napi::Error error = Napi::Error::New(info.Env(), e.what());
 		copy_object(env, e.m_value, error);
-		throw error;
```
Comment (blagoev) [inline src/node/node_class.hpp:331]: this is a mistake I guess

File: tests/spec/helpers/realm-logging.js [hunk 3]
```diff
@@ -27,11 +27,11 @@ function patch(Realm) {
         const originalClearTestState = Realm.clearTestState;
         Realm.clearTestState = function patchedClearTestState() {
             originalClearTestState.call(Realm);
-            setupLogging(Realm);
+            //setupLogging(Realm);
```
Comment (blagoev) [inline tests/spec/helpers/realm-logging.js:30]: if not needed please remove the commented code

File: tests/spec/helpers/realm-logging.js [hunk 4]
```diff
@@ -27,11 +27,11 @@ function patch(Realm) {
...
         }
 
         // Setup the logger initially
-        setupLogging(Realm);
+        //setupLogging(Realm);
```
Comment (blagoev) [inline tests/spec/helpers/realm-logging.js:34]: if not needed please remove the commented code

**Review by blagoev — CHANGES_REQUESTED** [review]

File: tests/spec/helpers/realm-logging.js (hunk 3 above)
Comment (kneth) [reply tests/spec/helpers/realm-logging.js:30]: I will add a FIXME; we need to figure out a way to enable general logging.

File: tests/spec/helpers/realm-logging.js (hunk 4 above)
Comment (kneth) [reply tests/spec/helpers/realm-logging.js:34]: I will add a FIXME; we need to figure out a way to enable general logging.

File: src/node/node_class.hpp (hunk 2 above)
Comment (kneth) [reply src/node/node_class.hpp]: Probably a left-over from a debugging session. I will revert it.

**Review by blagoev — COMMENTED** [review]
why are there so much changes in node_class.hpp. can we revert those

**Review by blagoev — APPROVED** [review]

**Review by blagoev — CHANGES_REQUESTED** [review]
I don't see the revert for throwing the error

**Review by blagoev — APPROVED** [review]

**Pull request merged**